### ISSUE TITLE
`MixtureSameFamily` outputting incorrect log probs

### ISSUE THREAD

**Issue by adamgayoso** [pyro-ppl/numpyro#1489]
```python
def gmm_log_prob(N=5000):
    mix = Categorical(logits=jnp.ones((N, 2)),)
    comp = Normal(jnp.zeros((N, 2)), jnp.ones((N,2)),)
    gmm = MixtureSameFamily(mix, comp,)
    data = jnp.zeros((N,))
    return gmm.log_prob(data),  Normal(0, 1).log_prob(data)
```

```
gmm_log_prob()
(DeviceArray([0.77420866, 0.77420866, 0.77420866, ..., 0.77420866,
              0.77420866, 0.77420866], dtype=float32),
 DeviceArray([-0.9189385, -0.9189385, -0.9189385, ..., -0.9189385,
              -0.9189385, -0.9189385], dtype=float32))
```

Distrax and torch provide the correct output, which should be equal here

**Comment by fehiepsi** [comment]
Thanks for catching this! It looks like we didn't [normalize the logits](https://github.com/pyro-ppl/numpyro/blob/f81ce325de402d9a9891497136a36e3b44f982cc/numpyro/distributions/mixtures.py#L275). Do you want to submit a fix? something like
```
     return jax.nn.log_softmax(self.mixing_distribution.logits) + component_log_probs
```
